zebra-state/src/service/finalized_state/disk_format/scan.rs

```diff
@@ -163,7 +163,8 @@ impl IntoDisk for SaplingScannedResult {
 
 impl FromDisk for SaplingScannedResult {
     fn from_bytes(bytes: impl AsRef<[u8]>) -> Self {
-        SaplingScannedResult(bytes.as_ref().try_into().unwrap())
+        // TODO: Change of confirm the `unwrap_or` is good enough.
+        SaplingScannedResult(bytes.as_ref().try_into().unwrap_or([0; 32]))
```

Comment:
I am not sure about this.

Reply:
I think this might cause the control block transaction ID problem you were concerned about here:
https://github.com/ZcashFoundation/zebra/pull/8080#discussion_r1421816842

Instead, deserialize `Option<SaplingScannedResult>`, which will correctly turn an empty transaction ID into `None`.

It might help to document that solution here using `expect()`, so we know what to do next time.

Reply:
This implementation is removed in:
https://github.com/ZcashFoundation/zebra/pull/8075/commits/105d29c4c01aa43ac6941b186ee1828a2ad26c3a